src/glmtools.jl

```diff
@@ -419,6 +431,10 @@ function devresid(::Binomial, y, μ::Real)
     end
 end
 devresid(::Gamma, y, μ::Real) = -2 * (log(y / μ) - (y - μ) / μ)
+function devresid(::Geometric, y, μ::Real)
+    v = 2 * (xlogy(y, y / μ) + xlogy(y + 1, (μ + 1) / (y + 1)))
+    return μ == 0 ? oftype(v, NaN) : v
```

Comment:
I would probably move this check above to shortcircuit faster when mu==0.

Reply:
Right - I will update the code similar to the following:
`μ == 0 ? NaN : 2 * (xlogy(y, y / μ) + xlogy(y + 1, (μ + 1) / (y + 1)))`

Reply:
```julia
   μ == 0 && return oftype(μ, NaN)
   return 2 * (xlogy(y, y / μ) + xlogy(y + 1, (μ + 1) / (y + 1)))
```

is probably how I would do it, but it's really a matter of personal style and what @nalimilan thinks :smile:

Reply:
I agree it sounds better to adapt the `NaN` type to that of the inputs. But I'm not sure whether it should be the type of `μ`, that of `y`, or `promote_type(typeof(μ), typeof(y))`...

Reply:
`promote_type(typeof(μ), typeof(y))` seems the way to go. Given that we have an expression with `y/mu` and `xlogy` never returns a narrower type than the inputs, we can expect the result to be at least as wide as `y/mu`.